pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2172,7 +2180,9 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 				listenerFilters = append(listenerFilters, filter)
 			}
 		}
-		match := &listener.FilterChainMatch{}
+		match := &listener.FilterChainMatch{
+			//DestinationPort: &wrappers.UInt32Value{Value: opts.port},
```

Comment:
Sorry I should revert. This is my initial attempt.
The idea is that we need to add dstPort for outbound listeners as per title. However, this function is used by inbound listeners, httproxy listeners as well. If I override the port here some unrelated tests would fail. 
Also this buildListener is doing the aggregation, we should be conservative to hide the magic here.